fix(genapi): zed (#4655)
scaleway/docs-content: pages/generative-apis/reference-content/adding-ai-to-zed-ide.mdx

---
meta:
  title: Adding AI to the Zed IDE using Generative APIs
  description: Learn how to integrate AI-powered code models into Zed using Scaleway's Generative APIs.
content:
  h1: Adding AI to to the Zed IDE using Generative APIs
  paragraph: Elevate your coding experience by integrating AI-powered code models into the Zed IDE. With Continue and Scaleway's Generative APIs, you can leverage AI to understand, generate, and optimize code with ease.
tags: generative-apis ai machine-learning language-models code-assistance vs-code continue

> validation_date: 2025-03-18

Reviewer: I don't think this is the right format for the dates, it's supposed to be

dates:
    validation: xx
    posted: xx

>   validation_date: 2025-03-18

Reviewer: ```suggestion
  validation: 2025-03-18
```

>   posted_date: 2025-03-18

Reviewer: ```suggestion
  posted: 2025-03-18
```

> Zed is an IDE (Integrated Development Environment) including AI coding assistance support. Scaleway's Generative APIs supports Zed AI code completion and more.

Reviewer: ```suggestion
Zed is an IDE (Integrated Development Environment) including AI coding assistance support. Scaleway's Generative APIs support Zed AI code completion and more.
```

- A valid [API key](/iam/how-to/create-api-keys/) for API authentication
- Installed [Zed](https://zed.dev/) on your local machine

### Configure custom endpoints and models

1. Edit Zed settings located in `settings.json`, and add the following content to it:
    ```json
    {
    "language_models": {
        "openai": {
        "api_url": "https://api.scaleway.ai/v1",
        "available_models": [
            {
            "name": "qwen2.5-coder-32b-instruct",
            "display_name": "Qwen 2.5 Coder 32B",
            "max_tokens": 128000
            }
        ],
        "version": "1"
        }
    },
    "assistant": {
        "default_model": {
        "provider": "openai",
        "model": "qwen2.5-coder-32b-instruct"
        },
        "version": "2"
    }
    }
    ```

    This configuration will add a `qwen2.5-coder-32b-instruct` Scaleway hosted model available with the Zed `openai` provider, and use it as default model.

2. Open AI Assistant configuration by either using the command palette and typing `assistant: show configuration` or clicking on the bottom right **Assistant Panel** button and then **Assistant menu** in top right and finally **Configure**.

> 3. Scroll down to the OpenAI configuration, and paste your Scaleway Secret Key as API Key credentials.

Reviewer: ```suggestion
3. Scroll down to the OpenAI configuration, and paste your Scaleway secret key as API Key credentials.
```

    <Message type="note">

>       Your key will be deleted if you restart Zed. To store it permanently, alternatively, you can also setup your Scaleway Secret Key as `OPENAI_API_KEY` environment variable and restart Zed.

Reviewer: ```suggestion
      Your key will be deleted if you restart Zed. To store it permanently, you can set up your Scaleway secret key as `OPENAI_API_KEY` environment variable and restart Zed.
```

    </Message>

> 4. Your setup is complete. If you open a new chat and select `Qwen 2.5 Coder 32B` model, you can send text and retrieve model answer. Additionally, you can also use [Inline Assist](https://zed.dev/docs/assistant/inline-assistant#:~:text=You%20can%20use%20ctrl%2Denter,even%20within%20the%20terminal%20panel.) feature when editing your code.

Reviewer: ```suggestion
4. Your setup is complete. If you open a new chat and select `Qwen 2.5 Coder 32B` model, you can send text and retrieve model answers. Additionally, you can use [Inline Assist](https://zed.dev/docs/assistant/inline-assistant#:~:text=You%20can%20use%20ctrl%2Denter,even%20within%20the%20terminal%20panel.) feature when editing your code.
```
